src/java/htsjdk/samtools/BAMIndexer.java

```diff
@@ -55,6 +55,10 @@
      */
     public BAMIndexer(final File output, final SAMFileHeader fileHeader) {
 
+        if (fileHeader.getSortOrder() != SAMFileHeader.SortOrder.coordinate) {
```

Comment:
@vdauwera in your experience with weird bams, have you seen many that don't specify their sort order but are actually coordinate sorted?  Should we worry about that case failing to index?  Rewriting the bam header is expensive I believe, so we may want to allow it.